internal/ca/helper.go

```diff
@@ -0,0 +1,197 @@
+package ca
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	clusterCAType = "cluster"
+	serviceCAType = "service"
+	typeKey       = "kmm.node.kubernetes.io/ca.type"
+)
+
+type ConfigMap struct {
+	KeyName string
+	Name    string
+}
+
+var errConfigMapNotFound = errors.New("ConfigMap not found")
+
+//go:generate mockgen -source=helper.go -package=ca -destination=mock_helper.go
+
+type Helper interface {
+	GetClusterCA(ctx context.Context, namespace string) (*ConfigMap, error)
+	GetServiceCA(ctx context.Context, namespace string) (*ConfigMap, error)
+	Sync(ctx context.Context, namespace string, owner client.Object) error
+}
+
+type helperImpl struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func NewHelper(client client.Client, scheme *runtime.Scheme) Helper {
+	return &helperImpl{
+		client: client,
+		scheme: scheme,
+	}
+}
+
+func (h *helperImpl) Sync(ctx context.Context, namespace string, owner client.Object) error {
+	logger := ctrl.LoggerFrom(ctx)
+
+	logger.Info("Syncing the Cluster CA ConfigMap")
+
+	clusterCM, err := h.getOrCreateClusterCAConfigMap(ctx, namespace)
+	if err != nil {
+		return fmt.Errorf("could not get the cluster CA ConfigMap: %v", err)
+	}
+
+	if err = h.syncConfigMapOwners(ctx, clusterCM, owner, logger); err != nil {
+		return fmt.Errorf("could not sync the cluster CA ConfigMap owners: %v", err)
+	}
+
+	logger.Info("Syncing the service CA ConfigMap")
+
+	serviceCM, err := h.getOrCreateServiceCAConfigMap(ctx, namespace)
+	if err != nil {
+		return fmt.Errorf("could not get the service CA ConfigMap: %v", err)
+	}
+
+	if err = h.syncConfigMapOwners(ctx, serviceCM, owner, logger); err != nil {
+		return fmt.Errorf("could not sync the service CA ConfigMap owners: %v", err)
+	}
+
```

Comment:
Since we log an info message that syncing has started, do we also want an info log to indicate that syncing is successful?